File: frontend_code/send.py
#!/usr/bin/env python
import pika
import sys
import json

thisList = []
for item in sys.argv[1:]:
	thisList.append(item)

data = {'operation': 'registration','username': thisList[0],'password': thisList[1],'email': thisList[2]}	

message = json.dumps(data)


cred = pika.PlainCredentials('kevin','kevin')

connection = pika.BlockingConnection(
    pika.ConnectionParameters(host='192.168.0.105', credentials=cred, virtual_host='authentication'))
channel = connection.channel()

channel.queue_declare(queue='hello')

channel.basic_publish(exchange='', routing_key='hello', body=message)
print(" [x] Sent 'Hello World!'")
connection.close()

File: frontend_code/rabbitmq/send.php
<?php

require_once '../vendor/autoload.php';
use PhpAmqpLib\Connection\AMQPStreamConnection;
use PhpAmqpLib\Message\AMQPMessage;

function publishMessage($JSON)
{
	#cred: ip-address, port, username, password, vhost
	$connection = new AMQPStreamConnection('192.168.0.105', 5672, 'kevin', 'kevin', 'authentication');
	$channel = $connection->channel();
	$channel->queue_declare('hello', false, false, false, false);
	$msg = new AMQPMessage($JSON);
	$channel->basic_publish($msg, '', 'hello');
	
	echo " [x] Sent 'Hello World!'\n";
	$channel->close();
	$connection->close();
}


function produceMessage($JSON, $VHOST, $QUEUE)
{
	#cred: ip-address, port, username, password, vhost
	$connection = new AMQPStreamConnection('192.168.0.105', 5672, 'kevin', 'kevin', $VHOST);
	$channel = $connection->channel();
	$channel->queue_declare($QUEUE, false, false, false, false);
	$msg = new AMQPMessage($JSON);
	$channel->basic_publish($msg, '', $QUEUE);
	
	$channel->close();
	$connection->close();
}

/*
function produceMessage($JSON, $VHOST, $QUEUE)
{
	#cred: ip-address, port, username, password, vhost
	$connection = new AMQPStreamConnection('localhost', 5672, 'kevin', 'kevin', $VHOST);
	$channel = $connection->channel();
	$channel->queue_declare($QUEUE, false, false, false, false);
	$msg = new AMQPMessage($JSON);
	$channel->basic_publish($msg, '', $QUEUE);
	
	$channel->close();
	$connection->close();
}
*/

?>
